template.tpl

```diff
@@ -692,8 +722,30 @@ const spDefaultCustomDefs = {
 
 
 // Helpers
+const determinateIsLoggingEnabled = () => {
+  if (!data.logType) {
+    return getContainerVersion().debugMode;
+  }
+  if (data.logType === 'no') {
+    return false;
+  }
+  if (data.logType === 'debug') {
+    return getContainerVersion().debugMode;
+  }
+
+  return data.logType === 'always';
+};
+
 const fail = (msg) => {
-  log('[ERROR GTM-SS / Snowplow Tag] ' + msg);
+  if (determinateIsLoggingEnabled()) {
+    log(JSON.stringify({
+      'Name': 'Snowplow',
+      'Type': 'Message',
+      'TraceId': getRequestHeader('trace-id'),
```

Comment:
This id is auto-generated by Google Template Gallery. If you install a template from Google Template Galery then you will get this id inside the template.

Reply:
I think yes. That's because technically if you have the 'trace-id' header. You will always have it because usually, it added by your load balancer on the first hop.

And you will be able to get benefits from these standardized logs in case you parse them with the proper tool. In other cases, there is no reason to leave this field empty or add a random identifier.